composer/trainer/trainer.py

```diff
@@ -140,6 +246,140 @@ def _unpack_evaluators(
     return evaluators
 
 
+def _unpack_grad_accum(grad_accum: Union[int, str], device: Device) -> Tuple[int, bool]:
+    # Set initial grad_accum to 1 if using adaptive
+    adaptive_grad_accum = grad_accum == "auto"
+    if adaptive_grad_accum:
+        grad_accum = 1
+        warnings.warn(textwrap.dedent("""Setting `grad_accum='auto'` is an experimental feature
+            which may cause uncaught Cuda Out of Memory errors. In this case, please manually
+            set grad_accum explicitly to an integer instead.
+            """),
+                      category=UserWarning)
+    # Cannot use adaptive grad accum on CPU
+    if isinstance(device, DeviceCPU) and adaptive_grad_accum:
+        raise ValueError("Cannot use adaptive grad_accum on CPU. Please set grad_accum >= 1")
```

Comment:
why not? Understand that there would never be a CUDA OOM, but allowthing this make its easier for folks to swtich between testing on CPU vs GPU. Maybe we allow this ,and this should be a no-op? cc: @mvpatel2000

Reply:
I'm not too sure about the consequences of an OOM on CPU. (The oom would be determined by the amount of RAM or the k8s limit). I think K8s automatically kills processes if they try to use too much ram, so there would be nothing to recover, but am not entirely sure. Anyways, tracking this feature request [here](https://mosaicml.atlassian.net/browse/CO-295).